src/Video.tsx

```diff
@@ -606,14 +611,75 @@ const Video = forwardRef<VideoRef, ReactVideoProps>(
         : ViewType.SURFACE;
     }, [drm, useSecureView, useTextureView, viewType]);
 
+    const _renderPoster = useCallback(() => {
+      // poster resize mode
+      let _posterResizeMode: ImageResizeMode = 'contain';
+
+      if (!isPosterDeprecated && poster?.resizeMode) {
+        _posterResizeMode = poster.resizeMode;
+      } else if (posterResizeMode && posterResizeMode !== 'none') {
+        _posterResizeMode = posterResizeMode;
+      }
+
+      // poster style
+      const baseStyle: StyleProp<ImageStyle> = {
+        ...StyleSheet.absoluteFillObject,
+        resizeMode: _posterResizeMode,
+      };
+
+      let posterStyle: StyleProp<ImageStyle> = baseStyle;
+
+      if (!isPosterDeprecated && poster?.style) {
+        const styles = Array.isArray(poster.style)
+          ? poster.style
+          : [poster.style];
+        posterStyle = [baseStyle, ...styles];
+      }
+
+      // render poster
+      if (renderLoader && poster) {
+        console.warn(
+          'You provided both `renderLoader` and `poster` props. `Poster` props is ignored.',
+        );
+      }
+
+      if (renderLoader) {
+        // check if valid jsx
+        if (React.isValidElement(renderLoader)) {
+          console.warn(
+            'Invalid renderLoader component. Please provide a valid JSX component',
+          );
+          return;
+        }
+
+        return <View style={StyleSheet.absoluteFill}>{renderLoader}</View>;
+      }
+
+      return (
+        <Image
+          {...(isPosterDeprecated ? {} : poster)}
+          source={isPosterDeprecated ? {uri: poster} : poster?.source}
+          style={posterStyle}
+        />
+      );
+    }, [isPosterDeprecated, poster, posterResizeMode, renderLoader]);
+
+    const _style: StyleProp<ViewStyle> = useMemo(
+      () => ({
+        ...StyleSheet.absoluteFillObject,
+        ...(showPoster ? {display: 'none'} : {}),
+      }),
+      [showPoster],
+    );
+
     return (
       <View style={style}>
         <NativeVideoComponent
           ref={nativeRef}
           {...rest}
           src={src}
           drm={_drm}
-          style={StyleSheet.absoluteFill}
+          style={_style}
```

Comment:
But here you put display='none' on the video view ? It is weird ?

Reply:
Then _style would not be needed anymore !

Reply:
I added `display: none` to prevent an issue that occurs when someone uses the poster. Sometimes, there is a situation where the video isn't fully loaded (it doesn't trigger `onReadyForDisplay`), causing a brief moment where both the first video slide and the poster are visible simultaneously, which shouldn't happen. An alternative approach could be to hide the poster in the `onVideoLoad` function instead of `_onReadyForDisplay`. However, this has its drawback as well, because the poster is visible when the video isn't ready for viewing yet (although it is loaded).
Let me please know which option is better for us

Reply:
OK, I understand (and I already fight with that issue).
please fix conflicts from this PR, I will test it myself.
Thank you